File: OOP-Encapsulation_RobertProdanov/src/Student.java
import java.util.ArrayList;

public class Student {

    private String name;
    private int age;
    private ArrayList<String> courses;

    public Student(String name, int age, ArrayList<String> courses) {
        this.name = name;
        this.age = age;
        this.courses = courses;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public ArrayList<String> getCourses() {
        return courses;
    }

    public void addCourse(String course){
        this.getCourses().add(course);
    }


    public void removeCourse(String course){
        this.getCourses().remove(course);

    }


}

File: OOP_Encapsulation_VladislavMilchov/Student.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace StudentCoursesEncapsulation
{
    internal class Student
    {

        private string name;
        private int age;
        private List<string> courses = new List<string>();

        public string Name
        {
            get { return name; } set { name = value; }
        }
        public int Age
        {
            get { return age; }
            set { age = value; }
        }
        public List<string> Courses
        {
            get { return courses; }
            set { courses = value; }
        }

        public Student()
        {
        }

        public Student(string name, int age)
        {
            this.name = name;
            this.age = age;

        }



        public void AddCourse(string course)
        {
            courses.Add(course);
        }

        public void RemoveCourse(string course)
        {
            if (courses.Contains(course))
            {
                courses.Remove(course);
            }
        }

        public void GetCourses()
        {

            for (int i = 0; i < courses.Count; i++)
            {                
                Console.WriteLine(courses[i]);
            }
        }

        public void GetAge()
        {
            Student student = new Student();
            student.Age = age;
            Console.WriteLine(student.age);
        }


        public void GetName()
        {
            Student student = new Student();
            student.Name = name;
            Console.WriteLine(student.name);

        }



        
        
    }
}
